controller/slot_controller.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+require_relative '../model/slot'
+require_relative '../model/car'
+require_relative '../views/slot_view'
+
+# Controller Class for Slot
+class SlotController
+
+  DEFAULT_LOT_SIZE = 10
+  def initialize(size = DEFAULT_LOT_SIZE)
+    return unless Slot.count.zero?
+
+    size.times do
+      Slot.create
+    end
+  end
+
+  def park_car(registration_number)
+    car_id = (Car.where(registration_number:).first || Car.create(registration_number:))[:id]
+    slot = Slot.where(car_id: nil).order(:id).first
+
+    raise 'Parking Lot is Full!!' unless slot
+
+    slot.update(car_id:, entry_time: Time.now).save
```

Comment:
park_car_in_slot can be a method in slot model. Or the other way to go about this is `park_car` method in slot model, which takes care of finding an empty slot and park the car in that slot position.